src/core/server/plugins/plugin.ts

```diff
@@ -88,12 +92,12 @@ export class PluginWrapper<
    * @param plugins The dictionary where the key is the dependency name and the value
    * is the contract returned by the dependency's `setup` function.
    */
-  public async setup(setupContext: CoreSetup, plugins: TPluginsSetup) {
+  public async setup(setupContext: CoreSetup<TPluginsStart>, plugins: TPluginsSetup) {
     this.instance = this.createPluginInstance();
 
     this.log.info('Setting up plugin');
 
-    return await this.instance.setup(setupContext, plugins);
+    return this.instance.setup(setupContext, plugins);
```

Comment:
The method is async, awaiting the returning promise was unnecessary.

Reply:
You are right. We use `await` excessively, even when it's not necessary `async () => await do()`. It's a matter of convention. Although it's not mentioned in core/conventions.md, so discussable.

Reply:
The only subtle difference will be the actual place it would throw from if the promise is rejected.

Reply:
I don't see the difference. I have the same output
```
UNHANDLED PROMISE REJECTION Error: not found
    at mostInner (/Users/mikhailshustov/work/kibana/src.js:21:9)
    at inner (/Users/mikhailshustov/work/kibana/src.js:25:16)
    at outer (/Users/mikhailshustov/work/kibana/src.js:29:16)
```
for 
```js
async function mostInner() {
  throw new Error('not found');
}

async function inner() {
  return await mostInner();
}

async function outer() {
  return await inner();
}

outer();
process.on('unhandledRejection', error => {
  console.error('UNHANDLED PROMISE REJECTION', error);
});
```
and
```js
function mostInner() {
  new Error('not found');
}

async function inner() {
  return mostInner();
}

async function outer() {
  return await inner();
}

outer();
process.on('unhandledRejection', error => {
  console.error('UNHANDLED PROMISE REJECTION', error);
});

```

Reply:
Yea you are right, tried it too this weekend. This is catched at the higher level anyway and gets the full stack in any case